crates/usvg/src/writer.rs

```diff
@@ -863,7 +898,117 @@ fn conv_element(node: &Node, is_clip_path: bool, ctx: &mut WriterContext, xml: &
             if let Some(ref flattened) = text.flattened {
                 conv_element(flattened, is_clip_path, ctx, xml);
             } else {
-                log::warn!("Text must be flattened.");
+                xml.start_svg_element(EId::Text);
+
+                if !text.id.is_empty() {
+                    xml.write_id_attribute(&text.id, ctx);
+                }
+
+                xml.write_attribute("xml:space", "preserve");
+
+                match text.writing_mode {
+                    WritingMode::LeftToRight => {}
+                    WritingMode::TopToBottom => xml.write_svg_attribute(AId::WritingMode, "tb"),
+                }
+
+                match text.rendering_mode {
```

Comment:
You sure? At least [here](https://developer.mozilla.org/en-US/docs/Web/CSS/text-rendering) it says it should be `auto`? Not sure though.

Reply:
You sure? At least [here](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/text-rendering) it says it should be `auto`? Although it does mention that in this case legibility should be preferred.

Reply:
Hmm... kinda. We do not represent `auto` at all, because it wasn't needed. 
`optimizeLegibility` is de-facto the default value, so let's just ignore it. Most libraries do not support this property to begin with, so it's fine.

And maybe later we add a proper `auto` support.